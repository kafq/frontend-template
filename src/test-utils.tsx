import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

function renderWithHistory(
  ui: any,
  {
    route = '/', history = createMemoryHistory({ initialEntries: [route] }), ...options
  }: any = {},
) {
  return {
    ...rtlRender(<Router history={history}>{ui}</Router>, options),
    history,
  };
}

export { renderWithHistory }
