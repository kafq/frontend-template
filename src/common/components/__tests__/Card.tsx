import React from 'react';

import { render } from '../../../test-utils';
import { Card } from '../';

it('renders Card component correctly', () => {
  const { container } = render(<Card title="Hello World"/>)
  expect(container.firstChild).toMatchSnapshot()
})