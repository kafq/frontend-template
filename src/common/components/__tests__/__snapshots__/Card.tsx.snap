// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component correctly 1`] = `
.emotion-0 {
  padding: 1rem 0.5rem;
  margin: 1rem;
  min-height: 300px;
  min-width: 180px;
  border: 1px solid #cdcdcd;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <h2>
    Hello World
  </h2>
</div>
`;
